### PR TITLE
Added Cmake and main as well as dummy include

### PR DESCRIPTION
**Pull request by Gaspard--**
This commit is here to define the initial layout of the repository and handle building.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -W -fdiagnostics-color=always -Wall -Wcast-align -Wcast-qual -Wconversion -Wctor-dtor-privacy -Wdisabled-optimization -Wduplicated-cond -Wextra -Wformat=2 -Winit-self -Wlogical-op -Wmissing-declarations -Wmissing-include-dirs -Wno-sign-conversion -Wnoexcept -Wnull-dereference -Wold-style-cast -Woverloaded-virtual -Wredundant-decls -Wstrict-aliasing=1 -Wstrict-null-sentinel -Wstrict-overflow=5 -Wswitch-default -Wno-unknown-pragmas -Wuseless-cast -Wno-parentheses -Wno-unused-local-typedefs")
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -stdlib=libc++ -Wno-missing-braces -std=c++17")
+endif()
+set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG_LAYERS")
+
+file(GLOB_RECURSE SOURCES_FILE "${SOURCE_DIRECTORY}/*.cpp")
+file(GLOB_RECURSE HEADERS_FILE "${HEADER_DIRECTORY}/*.hpp")
+
+find_package(Vulkan REQUIRED)
```
Comment (Zeng-Zer) [inline CMakeLists.txt:22]: Having some troubles with vulkan: `Could NOT find Vulkan (missing: Vulkan_INCLUDE_DIR)`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+- magma
+
+Pass `CLAWS_DIR` and `MAGMA_DIR` so that `cmake` finds them.
+
+Ex:
+```bash
+mkdir build/
+mkdir build/Debug
+cd build/Debug
+cmake ../.. -DCMAKE_BUILT_TYPE=Debug -DCLAWS_DIR=<claws dir> -DMAGMA_DIR<magma dir>
```
Comment (Zeng-Zer) [inline README.md]: `-DMAGMA_DIR<magma dir>` missing `=`

File: CMakeLists.txt (hunk 1 above)
Comment (Gaspard--) [reply CMakeLists.txt:22]: Do you vulkan-dev(el) installed?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,20 @@
...
+Dependencies:
+- vulkan-hpp
+- claws
+- magma
+
+Pass `CLAWS_DIR` and `MAGMA_DIR` so that `cmake` finds them.
+
+Ex:
+```bash
+mkdir build/
```
Comment (Zeng-Zer) [inline README.md]: Could be replaced with `mkdir build/Debug -p` directly

File: CMakeLists.txt (hunk 1 above)
Comment (Zeng-Zer) [reply CMakeLists.txt:22]: Fixed the problem, thx

File: README.md [hunk 4]
```diff
@@ -1,2 +1,19 @@
...
+- claws
+- magma
+
+Pass `CLAWS_DIR` and `MAGMA_DIR` so that `cmake` finds them.
+
+Ex:
+```bash
+mkdir -p build/Debug
+cd build/Debug
+cmake ../.. -DCMAKE_BUILT_TYPE=Debug -DCLAWS_DIR=<claws dir> -DMAGMA_DIR=<magma dir>
```
Comment (Zeng-Zer) [inline README.md]: Had some troubles with relative paths, but it works with absolute paths for Claws and Magma directories

Comment (Gaspard--) [reply README.md]: I'll clarify that.

**Review by Zeng-Zer — COMMENTED** [review]
Have troubles compiling, have this error message: `[..]/magma/include/magma/VulkanHandler.hpp:3:10: fatal error: vulkan/vulkan.hpp: No such file or directory`

**Comment by Gaspard--** [comment]
I guess you're missing vulkan-hpp.

**Comment by Gaspard--** [comment]
Added links to make it clearer where to find what.

**Review by Zeng-Zer — APPROVED** [review]

**Pull request merged**